core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -142,13 +158,14 @@ private static StaticDataTask.Row convertPartition(Partition partition) {
   private static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
     Types.StructType partitionType = Partitioning.partitionType(table);
     PartitionMap partitions = new PartitionMap(partitionType);
-
-    try (CloseableIterable<ContentFile<?>> files = planFiles(scan)) {
-      for (ContentFile<?> file : files) {
+    try (CloseableIterable<ManifestEntry<? extends ContentFile<?>>> entries = planEntries(scan)) {
+      for (ManifestEntry<? extends ContentFile<?>> entry : entries) {
+        Snapshot snapshot = table.snapshot(entry.snapshotId());
+        ContentFile<?> file = entry.file();
```

Comment:
Let's check if the entry status is not EXISTING, otherwise its a bit meaningless.  And let's add a test to capture this case.

Reply:
This is not done yet, right?  (Something equivalent to ManifestReader.liveEntries())

Reply:
I just realized the entry file status can change from added to existing without change its data_file to snapshot-id mapping when following your thought. So this imply that we do not need to filter entry by its status and I also added a new unit test to cover on how rewrite manifests and expire snapshot will impact our last updated timestamp and snapshotId in partitions table